go-controller/pkg/controller/controller.go

```diff
@@ -139,6 +141,10 @@ func (c *controller[T]) startWorkers() error {
 }
 
 func (c *controller[T]) stop() {
+	if c.stopped {
+		return
+	}
+	c.stopped = true
```

Comment:
could we just set the `stopChan` to nil on stop and test for it?

Reply:
that feels like a bit less obvious way to say that controller is already stopped, but I can do that

Reply:
we save memory :stuck_out_tongue: 
we can add a comment if we feel it would be needed